mkt/search/indexers.py

```diff
@@ -375,6 +375,32 @@ def _locale_field_mapping(field, analyzer):
             return mapping
 
     @classmethod
+    def attach_trending_and_popularity_mappings(cls, mapping):
+        doc_type = cls.get_mapping_type_name()
+        new_properties = {}
+
+        # Add global popularity and popularity by region.
+        new_properties['popularity'] = {
+            'type': 'long', 'doc_values': True
+        }
+        for region in mkt.regions.ALL_REGION_IDS:
+            new_properties['popularity_%s' % region] = {
+                'type': 'long', 'doc_values': True
+            }
+
+        # Add global trending and trending by region.
+        new_properties['trending'] = {
+            'type': 'long', 'doc_values': True
+        }
+        for region in mkt.regions.ALL_REGION_IDS:
+            new_properties['trending_%s' % region] = {
+                'type': 'long', 'doc_values': True
+            }
+
+        mapping[doc_type]['properties'].update(new_properties)
+        return mapping
```

Comment:
Since we're normalizing this, these looks like they can be collapsed and put into a loop.